### ISSUE TITLE
Pyright does not recognize modifications to sys.path

### ISSUE THREAD

**Issue by RealSaltyFish** [microsoft/pyright#1729]
**Describe the bug**
When importing from a directory previously added to `sys.path` in the code, Pyright reports that the import could not be resolved.

**To Reproduce**
Create a directory `test` containing a file `main.py` and a sub-directory `foo`. Under `test/foo` create a file `bar.py` with the following contents:
```python
#!/usr/bin/env python3
a = 1
```

Write the following contents in `test/main.py`:
```python
#!/usr/bin/env python3
import sys, os
sys.path.insert(0, os.path.join(os.path.join(os.path.dirname(__file__), "foo")))
import bar
print(bar.a)
```

Running `python3 test/main.py` will correctly output `1`, but Pyright will falsely report that the import `bar` could not be resolved.

**Expected behavior**
Pyright should recognize that `sys.path` has been updated to include the directory `test/foo` and correctly import symbols from `test/foo/bar.py`.

**Screenshots or Code**
<img width="1501" alt="screenshot" src="https://user-images.githubusercontent.com/64259296/113544092-36f0d100-961a-11eb-800b-0276ec8ba6ff.png">

**VS Code extension or command-line**
I mainly use neovim with coc.nvim and coc-pyright, but this problem can also be reproduced in VS Code with Pyright 1.1.126 and default configurations.

**Comment by erictraut** [comment]
Pyright is a static type checker, so it will not know about changes you make dynamically to sys.path at runtime. You will need to configure any changes statically in pyright's configuration file to tell it about any such changes. For details, refer to [this documentation](https://github.com/microsoft/pyright/blob/master/docs/configuration.md) and look the "extraPaths".
